File: ruby/user_management_policy/lib/user_management/update_user.rb
require_relative 'user'
require_relative 'user_repository'

module UserManagement
  class UpdateUser
    def initialize(user_repository)
      @user_repository = user_repository
    end

    def call(user_id, email, display_name)
      existing = @user_repository.find_by_id(user_id)
      raise ArgumentError, "User not found: #{user_id.value}" if existing.nil?
      @user_repository.save(User.new(id: user_id, email: email, display_name: display_name))
    end
  end
end

File: python/user_management_policy/update_user.py
from user_management_policy.user import User
from user_management_policy.user_id import UserId
from user_management_policy.user_repository import UserRepository


def make_update_user(user_repository: UserRepository):
    def update_user(user_id: UserId, email: str, display_name: str) -> None:
        existing = user_repository.find_by_id(user_id)
        if existing is None:
            raise ValueError(f"User not found: {user_id.value}")
        user_repository.save(User(id=user_id, email=email, display_name=display_name))
    return update_user
